Runtime/Core/UnityVectorExtensions.cs

```diff
@@ -51,6 +64,51 @@ public static Vector3 ProjectOntoPlane(this Vector3 vector, Vector3 planeNormal)
         {
             return (vector - Vector3.Dot(vector, planeNormal) * planeNormal);
         }
+        
+        /// <summary>
```

Comment:
I thought that we agreed to put this in ShrinkablePolygon

Reply:
Not this one. I moved the other ones, but this one is pretty generic. It finds the intersection of 2 lines.